Swapped navigation to Jetpack Compose Destinations

rvenky125 opened the pull request:
I've downgraded the compose version to 1.2 in order to use the Jetpack Compose Destinations library. I've tried to make it as much scalable as possible. Let me know if I had violated the project structure or any other clean code. I'll correct and will create another pull request.

rvenky125 committed:
Downgraded the compose version to 1.2-rc and swapped the navigation library to Jetpack Compose Destinations

kasem-sm commented:
Hi, Thank you for your contribution. Looks like you dedicated your time entirely for this. I will for sure acknowledge your contribution at my social profiles. 

Also I will review this PR soon. Until then please [run spotless apply command](https://github.com/kasem-sm/SlimeKT#contribution-guide-and-contributors-%EF%B8%8F) to clean ups wildcard imports.

rvenky125 commented:
Thank you. I've some personal pending work to do. I'll look into the spotless evening.

rvenky125 commented:
I ran optimize imports before committing the repo. So there occurred multiple wild card imports. Is there any shortcut to get class imports?

rvenky125 committed:
Commit with Spotless apply

rvenky125 committed:
Small change in the name of the of a fun in BottomNavigator

kasem-sm commented on screen/ui-article-detail/src/main/java/kasem/sm/ui_detail/DetailVM.kt:
```diff
@@ -32,7 +32,7 @@ class DetailVM @Inject constructor(
 ) : ViewModel() {
 
-    private val articleId = savedStateHandle.get<String>("id")?.toInt() ?: -1
+    private val articleId = savedStateHandle.get<Int>("id") ?: -1
```
I remember that I was facing some weird issue here. The app would crash whenever I changed the type to Int. Maybe it was a random issue but I did not want to waste my time and hence haha. Great that it's not working.

rvenky125 replied:
Yeah, I had also faced that. But using destinations it worked 😊

kasem-sm replied:
Actually I did a type, "Great that it's now working!" 😃

kasem-sm commented on screen/ui-auth/src/main/java/kasem/sm/ui_auth/login/LoginScreen.kt:
```diff
@@ -7,29 +7,38 @@ package kasem.sm.ui_auth.login
 @Composable
 fun LoginScreen(
     viewModel: LoginVM,
-    onLoginSuccess: () -> Unit,
```
Great. A Lot of composable have been cleaned up. Thank You!

rvenky125 replied:
Glad you liked it

kasem-sm commented:
Btw I guess UI Tests needs to be rewritten na?

rvenky125 commented:
> Btw I guess UI Tests needs to be rewritten na?

I think, Yes, but now it's easy to write them. I mean we have abstraction there.

kasem-sm commented:
> > Btw I guess UI Tests needs to be rewritten na?
> 
> I think, Yes, but now it's easy to write them. I mean we have abstraction there.

Yup. Got it. I would do that so I can also get familiar with the Compose Destination library. I would merge this PR soon!

raamcosta commented on screen/ui-article-detail/src/main/java/kasem/sm/ui_detail/DetailVM.kt:
```diff
@@ -32,7 +32,7 @@ class DetailVM @Inject constructor(
 ) : ViewModel() {
 
-    private val articleId = savedStateHandle.get<String>("id")?.toInt() ?: -1
+    private val articleId = savedStateHandle.get<Int>("id") ?: -1
```
I heard about this PR. Thanks for using compose destinations 🙂
You should avoid accessing nav args via their string key. Instead, use “savedStateHandle.navArgs\<YourNavArgsClass\>().id” for example 🙂

rvenky125 replied:
Okay, thanks for the review, there are three occurrences like this in the this pr. I'll change them.

kasem-sm replied:
Ah. Your library made that straightforward. That's great!! I'm inspired as well as amazed by the potential of this library.

raamcosta replied:
It’s just that compose destinations abstracts you from having to know how the args are handled. Also there are special cases where it may do more than just get it from savedStateHandle, and you may be surprised on how the info is saved in there, so it’s safer to always use the APIs 🙂

kasem-sm replied:
Yup as I go through the changes in the code, I'm already realizing how this library can be helpful!!

kasem-sm commented:
@rvenky125 Found an issue.

https://user-images.githubusercontent.com/83884225/181865101-5d20f53a-ae1d-4f0f-847f-3b1d05fc7901.mp4

I'm debugging this issue too. Let me know when you find any fix.

rvenky125 committed:
Changes for reading args from savedStateHandle

rvenky125 commented:
> @rvenky125 Found an issue.
> 
>  screen-20220730-065545.2.mp4 
> I'm debugging this issue too. Let me know when you find any fix.

Oh, I'll check that now

kasem-sm commented:
> > @rvenky125 Found an issue.
> > screen-20220730-065545.2.mp4
> > I'm debugging this issue too. Let me know when you find any fix.
> 
> Oh, I'll check that now

I found the issue. Please refer to the pending review at top :) .

raamcosta commented on buildSrc/src/main/java/Compose.kt:
```diff
@@ -1,14 +1,12 @@
     private const val compostActivity = "1.6.0-alpha05"
-    private const val compostNavigation = "2.5.0-alpha01"
-    private const val composeVersion = "1.3.0-alpha01"
+    private const val composeVersion = "1.2.0-rc03"
```
I actually just today released a new version for compose 1.3.
So if you want, I believe you can keep this compose version and use 1.7.15-beta compose destinations.

rvenky125 committed:
Fixed bug with list not loading on topic click

rvenky125 committed:
Fixed bug with list not loading on topic click

rvenky125 commented:
> > > @rvenky125 Found an issue.
> > > screen-20220730-065545.2.mp4
> > > I'm debugging this issue too. Let me know when you find any fix.
> > 
> > 
> > Oh, I'll check that now
> 
> I found the issue. Please refer to the pending review at top :) .



> > > @rvenky125 Found an issue.
> > > screen-20220730-065545.2.mp4
> > > I'm debugging this issue too. Let me know when you find any fix.
> > 
> > 
> > Oh, I'll check that now
> 
> I found the issue. Please refer to the pending review at top :) .

Yeah, I fixed that and committed. Sorry, it's really a stupid thing I did.

kasem-sm commented:
No issues. You did a lot of refactoring in past few days. Great and Thank You again for this hard work.

kasem-sm commented:
Also there are lot of code violation when you run `gradle spotlessCheck`. Please run `gradle spotlessApply` to fix all those.

rvenky125 commented:
> Also there are lot of code violation when you run `gradle spotlessCheck`. Please run `gradle spotlessApply` to fix all those.

Is it? I've already run that. I just run again. I'm getting BUILD SUCCESSFUL. Can you pls check again with the latest commit?

rvenky125 commented:
I just realized that I've changed the way of using the backhander. Do you really need it as a dependency? If yes, I'll commit by implementing it.

rvenky125 commented:
> I just realized that I've changed the way of using the backhander. Do you really need it as a dependency? If yes, I'll commit by implementing it.

I've implemented that backhander as a dep. Let me if you wanted to be in the code.

rvenky125 commented:
> No issues. You did a lot of refactoring in past few days. Great and Thank You again for this hard work.

I'm thankful to you also. I've learnt how to implement compose destinations in Multi Module arch.

kasem-sm commented:
> > Also there are lot of code violation when you run `gradle spotlessCheck`. Please run `gradle spotlessApply` to fix all those.
> 
> Is it? I've already run that. I just run again. I'm getting BUILD SUCCESSFUL. Can you pls check again with the latest commit?

if You got Build Success, then check if there is any uncommitted files.

kasem-sm commented:
> > I just realized that I've changed the way of using the backhander. Do you really need it as a dependency? If yes, I'll commit by implementing it.
> 
> I've implemented that backhander as a dep. Let me if you wanted to be in the code.

Can you please share the file or line number?

kasem-sm commented:
Also replace the content of the file inside of `.github/workflows/android_action.yml` to [this](https://github.com/kasem-sm/SlimeKT/blob/dev/.github/workflows/android_action.yml) and push the changes so the code review process will start in Github Actions.

rvenky125 commented:
> > > I just realized that I've changed the way of using the backhander. Do you really need it as a dependency? If yes, I'll commit by implementing it.
> > 
> > 
> > I've implemented that backhander as a dep. Let me if you wanted to be in the code.
> 
> Can you please share the file or line number?

It's the HomeScreen composable. Where in previous you have backhander as function to it. But I've replaced it with direct import of BackHandler. Once check the changes in HomeScreen composable, You'll get an idea

rvenky125 committed:
Applied spotless apply, passing BackHandler as dependency to NavHost.

kasem-sm commented:
Thank you @rvenky125 for your valuable contribution to this repository.

rvenky125 commented:
> Thank you @rvenky125 for your valuable contribution to this repository.

It's my pleasure. It's my first contribution and you liked it. It's encouraging me to do more contributions.

kasem-sm commented:
Definitely. Also, Open source contributions can be added on your portfolio/resume as well and believe me, it helps a lot.

rvenky125 commented:
> Definitely. Also, Open source contributions can be added to your portfolio/resume as well and believe me, it helps a lot.

Yes, I learnt Android/React native development on my own. I just had my BSC and got a job as a junior SE in a startup company. Mostly it has React Native projects. I'm not that much joy with React native as Android. I wanna become experienced Android Developer. I feel very happy to take any career suggestions from you.

I'll try to contribute as much as possible besides doing my work ☺
